### PR TITLE
Cleanup atexit handling (cleaned version)

### PR DESCRIPTION
**Pull request by drmarkwslater**
This is another version of #510 without the horrible mess while I was tracking down a couple of lockups.

### CODE REVIEW

**Comment by drmarkwslater** [comment]
Any objections to merge?

File: python/Ganga/Core/InternalServices/ShutdownManager.py [hunk 1]
```diff
@@ -38,157 +38,110 @@
...
-
-    from Ganga.Core.GangaThread.WorkerThreads import shutDownQueues
-    shutDownQueues()
-
-    logger.info("Shutting Down Ganga Repositories")
-    from Ganga.Runtime import Repository_runtime
-    Repository_runtime.shutdown()
-
-    from Ganga.Core.InternalServices import Coordinator
+            logger.warning("Exception raised while stopping the monitoring: %s" % err)
```
Comment (milliams) [inline python/Ganga/Core/InternalServices/ShutdownManager.py]: For more detailed information (e.g. the full stacktrace), perhaps these should use [`logger.exception()`](https://docs.python.org/2.6/library/logging.html#logging.Logger.exception)?

Comment (drmarkwslater) [inline python/Ganga/Core/InternalServices/ShutdownManager.py]: Makes sense 😄

**Comment by drmarkwslater** [comment]
Anything more to change/add?

**Comment by milliams** [comment]
Looks good to me :+1:

**Pull request merged**